projects/ngx-maplibre-gl/src/lib/source/canvas-source.component.ts

```diff
@@ -1,95 +1,78 @@
 import {
   ChangeDetectionStrategy,
   Component,
-  Input,
   OnChanges,
-  OnDestroy,
-  OnInit,
   SimpleChanges,
+  inject,
+  input,
 } from '@angular/core';
-import { CanvasSource, CanvasSourceSpecification } from 'maplibre-gl';
-import { fromEvent, Subscription } from 'rxjs';
-import { filter } from 'rxjs/operators';
-import { MapService } from '../map/map.service';
+import type { CanvasSource, CanvasSourceSpecification } from 'maplibre-gl';
+import { SourceDirective as SourceDirective } from './source.directive';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { tap } from 'rxjs';
 
 /**
  * `mgl-canvas-source` - a canvas source component
  * @see [canvas](https://maplibre.org/maplibre-style-spec/sources/#canvas)
- * 
+ *
  * @category Source Components
  */
 @Component({
   selector: 'mgl-canvas-source',
   template: '',
   changeDetection: ChangeDetectionStrategy.OnPush,
+  hostDirectives: [{ directive: SourceDirective, inputs: ['id'] }],
   standalone: true,
 })
-export class CanvasSourceComponent
-  implements OnInit, OnDestroy, OnChanges, CanvasSourceSpecification {
-  /**  Init input */
-  @Input() id: string;
+export class CanvasSourceComponent implements OnChanges {
+  /** Init injections */
+  private readonly sourceDirective = inject(SourceDirective);
 
   /** Dynamic input */
-  @Input() coordinates: CanvasSourceSpecification['coordinates'];
-  /** Dynamic input */
-  @Input() canvas: CanvasSourceSpecification['canvas'];
-  /** Dynamic input */
-  @Input() animate?: CanvasSourceSpecification['animate'];
+  readonly coordinates =
+    input.required<CanvasSourceSpecification['coordinates']>();
 
-  type: CanvasSourceSpecification['type'] = 'canvas';
-  private sourceAdded = false;
-  private sub = new Subscription();
+  /** Dynamic input */
+  readonly canvas = input.required<CanvasSourceSpecification['canvas']>();
 
-  constructor(private mapService: MapService) {}
+  /** Dynamic input */
+  readonly animate = input<CanvasSourceSpecification['animate']>();
 
-  ngOnInit() {
-    const sub1 = this.mapService.mapLoaded$.subscribe(() => {
-      this.init();
-      const sub = fromEvent(this.mapService.mapInstance, 'styledata')
-        .pipe(filter(() => !this.mapService.mapInstance.getSource(this.id)))
-        .subscribe(() => {
-          this.init();
-        });
-      this.sub.add(sub);
-    });
-    this.sub.add(sub1);
+  constructor() {
+    this.sourceDirective.loadSource$
+      .pipe(
+        tap(() =>
+          this.sourceDirective.addSource(this.getCanvasSourceSpecification() as any)
```

Comment:
Why do we need this "as any"?

Reply:
CanvasSourceSpecification is not part of SourceSpecification